tools/update-go.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [[ $# -lt 1 ]]; then
+    echo "Usage: $0 $GO_VERSION"
+    echo
+    echo "Example: $0 1.21.1"
+    exit 2
+fi
+
+TOOLSDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+ROOTDIR=$(realpath "${TOOLSDIR}/..")
+
+VERSION="${1}"
+OUTPUT="${ROOTDIR}/hashes/go"
+PACKAGE_ROOT="https://go.dev/dl"
+
+# Get the go source package
+# e.g. https://go.dev/dl/go1.21.1.src.tar.gz
+GO_SRC_PACKAGE="go${VERSION}.src.tar.gz"
+GO_SRC_URL="${PACKAGE_ROOT}/${GO_SRC_PACKAGE}"
+
+curl -s -L -O -C - "${GO_SRC_URL}"
```

Comment:
nit: It would be nice to download this to a temporary directory so it doesn't leave the tar file somewhere on the user's system to clean up since it won't be used directly in the build.

Reply:
I kind of prefer it _not_ being in a temporary directory, just to make it easier to copy to our cache.

Reply:
That is a good point. As a rule I tend to get annoyed at tools that "do something nice for me but don't clean up their mess" but in this case, its not a mess, its a useful artifact for the next step in the process.

Reply:
IIRC, on another of these I changed it. I was originally downloading to `/tmp`, but on Patrick's suggestion I changed it to be local specifically to help with that next step. So I do generally agree and also don't like things being left around, but in this case there is some value to it.